Src/Common/FSecure/AsanaApi/AsanaApi.cpp

```diff
@@ -0,0 +1,196 @@
+		if (taskName.find(OBF(":writing")) == std::string::npos) { // make sure ':writing' is not in the task name
+			std::string taskId = task[OBF("gid")];
+			std::string creationTimestamp = task[OBF("created_at")];
+			std::vector<uint8_t> body;
+			bool dataFromAttachment;
+			if (task[OBF("attachments")].empty()) {
+				// No attachments in task, so the body is in the "notes" attribute
+				std::string notes = task[OBF("notes")];
+				body = std::vector<uint8_t>(std::make_move_iterator(notes.begin()), std::make_move_iterator(notes.end()));
+				dataFromAttachment = false;
+			} else {
+				// Attachment found! Body is in the attachment.
+				std::string attachmentId = task[OBF("attachments")][0][OBF("gid")];
+				body = GetAttachmentById(attachmentId);
+				dataFromAttachment = true;
+			}
+			ret.emplace_back(std::make_tuple(std::move(taskId), std::move(creationTimestamp), std::move(body), std::move(dataFromAttachment)));
+		}
+	}
+	return ret;
+}
+
+void FSecure::AsanaApi::DeleteTask(std::string const& taskId) {
+	std::string url = OBF("https://app.asana.com/api/1.0/tasks/") + taskId;
+	SendJsonRequest(url, NULL, Method::DEL);
+}
+
+std::string FSecure::AsanaApi::GetOrCreateSectionIdByName(std::string const& sectionName) {
+	// Check if the section already exists
+	std::string sectionId = GetSectionIdByName(sectionName);
+	if (sectionId.empty()) {
+		// If section does not exist, create it.
+		sectionId = CreateSectionIdByName(sectionName);
+	}
+	return sectionId;
+}
+
+std::string FSecure::AsanaApi::GetSectionIdByName(std::string const& sectionName) {
+	std::string url = OBF("https://app.asana.com/api/1.0/projects/") + this->m_ProjectId + OBF("/sections");
+	json response = SendJsonRequest(url, NULL, Method::GET);
+	for (auto& section : response[OBF("data")]) {
+		std::string name = section[OBF("name")];
+		if (name == sectionName) {
+			return section[OBF("gid")];
+		}
+	}
+	return "";
+}
+
+std::string FSecure::AsanaApi::CreateSectionIdByName(std::string const& sectionName) {
+	std::string url = OBF("https://app.asana.com/api/1.0/projects/") + this->m_ProjectId + OBF("/sections");
+	json body;
+	body[OBF("data")][OBF("name")] = sectionName;
+	json response = SendJsonRequest(url, body, Method::POST);
+	return response[OBF("data")][OBF("gid")];
+}
+
+std::string FSecure::AsanaApi::AddAttachmentToTask(std::string const& taskId, std::vector<uint8_t> const& attachmentBody, std::string const& attachmentFileName, std::string const& attachmentMimeType) {
+	std::string url = OBF("https://app.asana.com/api/1.0/tasks/") + taskId + OBF("/attachments");
+	// Generating body
+	const std::string boundary_prefix(OBF("------WebKitFormBoundary")); // Mimicking WebKit
+	const std::string alphabet(OBF("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"));
+	std::string boundary = boundary_prefix;
+	for (int i = 0; i < 16; i++) { // Generate random boundary string
+		boundary.push_back(alphabet[rand() % alphabet.size()]);
+	}
```

Comment:
Please use `boundary += Utils::GenerateRandomString(16)`

Reply:
Resolved in e8c2dca8466d567b598e828144e1e8f6f5e23bfd.